java/core/pom.xml

```diff
@@ -0,0 +1,194 @@
+                  <arg value="${test.proto.dir}/com/google/protobuf/non_nested_extension.proto" />
+                  <arg value="${test.proto.dir}/com/google/protobuf/non_nested_extension_lite.proto" />
+                  <arg value="${test.proto.dir}/com/google/protobuf/outer_class_name_test.proto" />
+                  <arg value="${test.proto.dir}/com/google/protobuf/outer_class_name_test2.proto" />
+                  <arg value="${test.proto.dir}/com/google/protobuf/outer_class_name_test3.proto" />
+                  <arg value="${test.proto.dir}/com/google/protobuf/test_bad_identifiers.proto" />
+                  <arg value="${test.proto.dir}/com/google/protobuf/test_check_utf8.proto" />
+                  <arg value="${test.proto.dir}/com/google/protobuf/test_check_utf8_size.proto" />
+                  <arg value="${test.proto.dir}/com/google/protobuf/test_custom_options.proto" />
+                  <arg value="${test.proto.dir}/com/google/protobuf/any_test.proto" />
+                  <arg value="${test.proto.dir}/com/google/protobuf/field_presence_test.proto" />
+                  <arg value="${test.proto.dir}/com/google/protobuf/map_for_proto2_lite_test.proto" />
+                  <arg value="${test.proto.dir}/com/google/protobuf/map_for_proto2_test.proto" />
+                  <arg value="${test.proto.dir}/com/google/protobuf/map_test.proto" />
+                  <arg value="${test.proto.dir}/com/google/protobuf/map_initialization_order_test.proto" />
+                </exec>
+              </tasks>
+            </configuration>
+            <goals>
+              <goal>run</goal>
+            </goals>
+          </execution>
+        </executions>
+      </plugin>
+
+      <!-- Add the generated sources to the build -->
+      <plugin>
+        <artifactId>maven-compiler-plugin</artifactId>
+        <configuration>
+          <generatedSourcesDirectory>${generated.sources.dir}</generatedSourcesDirectory>
+          <generatedTestSourcesDirectory>${generated.testsources.dir}</generatedTestSourcesDirectory>
+        </configuration>
+      </plugin>
+
+      <!-- OSGI bundle configuration -->
+      <plugin>
+        <groupId>org.apache.felix</groupId>
+        <artifactId>maven-bundle-plugin</artifactId>
+        <extensions>true</extensions>
+        <configuration>
+          <instructions>
+            <Bundle-DocURL>https://developers.google.com/protocol-buffers/</Bundle-DocURL>
+            <Bundle-SymbolicName>com.google.protobuf</Bundle-SymbolicName>
+            <Export-Package>com.google.protobuf;version=${project.version}</Export-Package>
+          </instructions>
+        </configuration>
+      </plugin>
+
+      <plugin>
+        <artifactId>maven-jar-plugin</artifactId>
+        <executions>
+
+          <!-- Configure a test jar artifact that can be used by other modules -->
+          <execution>
+            <id>test-jar</id>
```

Comment:
It will take some time to switch to gradle. There needs a grace period where we support both and eventually only gradle if we decide gradle is the right build solution. It doesn't help solve the immediate issue here.

I think we only live with another testing artifact as long as we make it clear it's only for protobuf testing. It seems grpc has a grpc-testing artifact. Can we name the protobuf as protobuf-java-testing?